src/NuGet.Core/NuGet.Indexing/SortExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.Indexing
+{
+    public static class SortExtensions
+    {
+        public static IEnumerable<T> Merge<T>(this IEnumerable<T> x, IEnumerable<T> y, IComparer<T> comparer)
+        {
+            if (x == null)
+            {
+                throw new ArgumentNullException("x");
+            }
+
+            if (y == null)
+            {
+                throw new ArgumentNullException("y");
+            }
+
+            if (comparer == null)
+            {
+                throw new ArgumentNullException("comparer");
+            }
+
+            var ex = x.GetEnumerator();
+            var ey = y.GetEnumerator();
+
+            var fx = ex.MoveNext();
+            var fy = ey.MoveNext();
+
+            while (true)
+            {
+                if (fx & fy)
+                {
+                    if (comparer.Compare(ex.Current, ey.Current) < 0)
```

Comment:
I've updated this to a <= which has the impact of returning the first list first if everything is equal. You could argue for other semantics - for example interleaving - however that would really be outside the scope of what Merge means. The idea is that behavior is strictly injected through the Comparer otherwise the behavior is transparent.